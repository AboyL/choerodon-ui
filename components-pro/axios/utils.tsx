import buildURL from 'axios/lib/helpers/buildURL';

export function buildSortedURL(...args: any[]) {
  const builtURL = buildURL(...args);

  const [urlPath, queryString] = builtURL.split('?');

  if (queryString) {
    const paramsPair = queryString.split('&');
    return `${urlPath}?${paramsPair.sort().join('&')}`;
  }

  return builtURL;
}

export function buildURLWithData(data, ...args: any[]) {
  const builtURL = buildSortedURL(...args);
  if (data) {
    return `${builtURL}|${JSON.stringify(data)}`;
  }
  return builtURL;
}

export function isCacheLike(cache: any) {
  return !!(
    cache.set &&
    cache.get &&
    cache.del &&
    typeof cache.get === 'function' &&
    typeof cache.set === 'function' &&
    typeof cache.del === 'function'
  );
}
