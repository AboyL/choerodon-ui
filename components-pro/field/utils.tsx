import isObject from 'lodash/isObject';
import isNil from 'lodash/isNil';
import { isArrayLike } from 'mobx';
import { FieldType } from '../data-set/enum';
import Constants from '../data-set/Constants';
import Field from '../data-set/Field';

export function toRangeValue(value: any, range?: boolean | [string, string]): [any, any] {
  if (isArrayLike(range)) {
    if (isObject(value)) {
      const [start, end] = range;
      return [value[start], value[end]];
    }
  } else if (isArrayLike(value)) {
    return value.slice(0, 2) as [any, any];
  }
  return [undefined, undefined];
}

export function fromRangeValue(value: any[], range?: boolean | [string, string]): any {
  if (isArrayLike(range)) {
    const [start, end] = range;
    return {
      [start]: value[0],
      [end]: value[1],
    };
  }
  return value;
}

export function toMultipleValue(value: any, range?: boolean | [string, string]) {
  if (!isNil(value)) {
    const multipleValue = isArrayLike(value) ? value.slice() : [value];
    if (range) {
      return multipleValue.map(item => toRangeValue(item, range));
    }
    return multipleValue;
  }
  return [];
}

export function getDateFormatByFieldType(type: FieldType) {
  switch (type) {
    case FieldType.date:
      return Constants.DATE_FORMAT;
    case FieldType.dateTime:
      return Constants.DATE_TIME_FORMAT;
    case FieldType.week:
      return Constants.WEEK_FORMAT;
    case FieldType.month:
      return Constants.MONTH_FORMAT;
    case FieldType.year:
      return Constants.YEAR_FORMAT;
    case FieldType.time:
      return Constants.TIME_FORMAT;
    default:
      return Constants.DATE_FORMAT;
  }
}

export function getDateFormatByField(field?: Field, type?: FieldType): string {
  if (field) {
    return field.get('format') || getDateFormatByFieldType(type || field.type);
  }
  if (type) {
    return getDateFormatByFieldType(type);
  }
  return Constants.DATE_JSON_FORMAT;
}
