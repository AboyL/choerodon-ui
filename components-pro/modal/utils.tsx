import { isValidElement, ReactNode } from 'react';
import { isArrayLike } from 'mobx';
import isNil from 'lodash/isNil';
import isBoolean from 'lodash/isBoolean';
import isNumber from 'lodash/isNumber';
import isString from 'lodash/isString';
import { ModalProps } from './Modal';

export type confirmProps = { iconType?: string; type?: string; children?: any };

export function normalizeProps(
  props: ModalProps & confirmProps | ReactNode,
): ModalProps & confirmProps {
  if (
    isBoolean(props) ||
    isNil(props) ||
    isNumber(props) ||
    isString(props) ||
    isValidElement(props) ||
    isArrayLike(props)
  ) {
    return {
      children: props,
    };
  }
  return props;
}
