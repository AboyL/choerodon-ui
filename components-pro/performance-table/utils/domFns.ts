// @ts-ignore
export function addEvent(el?: any, event: string, handler: EventListener | EventListenerObject | null, inputOptions?: Object): void {
  if (!el) return;
  const options = { capture: true, ...inputOptions };
  if (el.addEventListener) {
    el.addEventListener(event, handler, options);
  } else if (el.attachEvent) {
    el.attachEvent('on' + event, handler);
  } else {
    // $FlowIgnore: Doesn't think elements are indexable
    el['on' + event] = handler;
  }
}

export function removeEvent(el: any, event: string, handler: Function, inputOptions?: Object): void {
  if (!el) return;
  const options = { capture: true, ...inputOptions };
  if (el.removeEventListener) {
    el.removeEventListener(event, handler, options);
  } else if (el.detachEvent) {
    el.detachEvent('on' + event, handler);
  } else {
    // $FlowIgnore: Doesn't think elements are indexable
    el['on' + event] = null;
  }
}
