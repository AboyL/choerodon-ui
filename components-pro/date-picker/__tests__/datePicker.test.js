import React from 'react';
import { mount } from 'enzyme';
import moment from 'moment';
import MockDate from 'mockdate';
import DatePicker from '..';
import DatePickerTest from './DatePicker';
import { openPanel, clearInput, nextYear, nextMonth } from './utils';
import focusTest from '../../../tests/shared/focusTest';

describe('date-picker-pro', () => {
  focusTest(DatePicker);

  beforeEach(() => {
    MockDate.set(moment('2020-01-22'));
  });

  afterEach(() => {
    MockDate.reset();
  });

  it('renders datePicker correctly', () => {
    const wrapper = mount(<DatePicker />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders multiple correctly', () => {
    const wrapper = mount(<DatePicker defaultValue="2020-02-10 18:18:18" />);
    expect(
      wrapper
        .find('input')
        .at(0)
        .prop('value'),
    ).toBe('2020-02-10');
    wrapper.setProps({ multiple: true });
    wrapper.update();
    expect(
      wrapper
        .find('.c7n-pro-calendar-picker-wrapper')
        .at(0)
        .hasClass('c7n-pro-calendar-picker-multiple'),
    ).toBe(true);
  });

  it('renders dataSet multiple correctly', () => {
    const wrapper = mount(<DatePickerTest />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('DatePicker').props().name).toEqual('date');
    expect(
      wrapper
        .find('.c7n-pro-calendar-picker-wrapper')
        .at(0)
        .hasClass('c7n-pro-calendar-picker-multiple'),
    ).toBe(true);
    expect(
      wrapper
        .find('.c7n-pro-calendar-picker-range-end')
        .at(0)
        .prop('value'),
    ).toBe('2020-02-09');
    expect(
      wrapper
        .find('.c7n-pro-calendar-picker-range-start')
        .at(0)
        .prop('value'),
    ).toBe('1984-11-22');
  });

  it('the onChange event only triggers when date was selected or change', () => {
    const handleChange = jest.fn();
    const wrapper = mount(<DatePicker onChange={handleChange} />);
    openPanel(wrapper);
    nextYear(wrapper);
    expect(handleChange).not.toHaveBeenCalled();
    nextMonth(wrapper);
    expect(handleChange).not.toHaveBeenCalled();
  });

  it('the clear button will clear the input value', () => {
    const wrapper = mount(<DatePicker value={moment('2019-11-23')} />);
    expect(
      wrapper
        .find('input')
        .at(0)
        .prop('value'),
    ).toBe('2019-11-23');
    clearInput(wrapper);
    expect(
      wrapper
        .find('input')
        .at(0)
        .prop('value'),
    ).toBe('');
  });
});
