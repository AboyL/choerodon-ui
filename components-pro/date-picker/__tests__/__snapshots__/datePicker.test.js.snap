// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date-picker-pro renders dataSet multiple correctly 1`] = `
<DatePickerTest>
  <DatePicker
    autoComplete="off"
    clearButton={true}
    dataSet={
      DataSet {
        "cachedSelected": Array [],
        "children": Object {},
        "currentPage": 1,
        "dataToJSON": "dirty",
        "el": undefined,
        "events": Object {
          "indexchange": Array [
            Array [
              [Function],
              false,
            ],
          ],
          "update": Array [
            Array [
              [Function],
              false,
            ],
          ],
        },
        "fields": Object {
          "date": Field {
            "dataSet": [Circular],
            "isDynamicPropsComputing": false,
            "lastDynamicProps": Object {},
            "pending": PromiseQueue {
              "queue": Array [],
              "queueing": false,
            },
            "pristineProps": Object {
              "defaultValue": Object {
                "end": "2020-02-09",
                "start": "1984-11-22",
              },
              "name": "date",
              "range": Array [
                "start",
                "end",
              ],
              "required": true,
              "type": "date",
              "validator": [Function],
            },
            "props": Object {
              "defaultValue": Object {
                "end": "2020-02-09",
                "start": "1984-11-22",
              },
              "name": "date",
              "range": Array [
                "start",
                "end",
              ],
              "required": true,
              "type": "date",
              "validator": [Function],
            },
            "record": undefined,
            "validator": Validator {
              "control": undefined,
              "field": [Circular],
              "innerValidationResults": Array [],
            },
          },
        },
        "inBatchSelection": false,
        "name": undefined,
        "originalData": Array [],
        "pageSize": 10,
        "parent": undefined,
        "parentName": undefined,
        "pending": PromiseQueue {
          "queue": Array [],
          "queueing": false,
        },
        "props": Object {
          "autoCreate": true,
          "autoLocateAfterCreate": true,
          "autoLocateAfterRemove": true,
          "autoLocateFirst": true,
          "autoQuery": false,
          "autoQueryAfterSubmit": true,
          "dataToJSON": "dirty",
          "events": Object {
            "update": [Function],
          },
          "fields": Array [
            Object {
              "defaultValue": Object {
                "end": "2020-02-09",
                "start": "1984-11-22",
              },
              "name": "date",
              "range": Array [
                "start",
                "end",
              ],
              "required": true,
              "type": "date",
              "validator": [Function],
            },
          ],
          "modifiedCheck": true,
          "pageSize": 10,
          "paging": true,
          "selection": "multiple",
        },
        "queryParameter": Object {},
        "records": Array [
          Record {
            "cascadeRecordsMap": Object {},
            "data": Object {
              "date": Object {
                "end": "2020-02-08T16:00:00.000Z",
                "start": "1984-11-21T16:00:00.000Z",
              },
            },
            "dataSet": [Circular],
            "dataSetSnapshot": Object {},
            "editing": undefined,
            "fields": Object {
              "date": Field {
                "dataSet": [Circular],
                "isDynamicPropsComputing": false,
                "lastDynamicProps": Object {},
                "pending": PromiseQueue {
                  "queue": Array [],
                  "queueing": false,
                },
                "pristineProps": Object {
                  "name": "date",
                },
                "props": Object {
                  "name": "date",
                },
                "record": [Circular],
                "validator": Validator {
                  "control": undefined,
                  "field": [Circular],
                  "innerValidationResults": Array [],
                },
              },
            },
            "id": 1001,
            "isCached": false,
            "isCurrent": true,
            "isSelected": false,
            "pristineData": Object {
              "date": Object {
                "end": "2020-02-08T16:00:00.000Z",
                "start": "1984-11-21T16:00:00.000Z",
              },
            },
            "selectable": true,
            "state": Object {},
            "status": "add",
          },
        ],
        "resetInBatch": false,
        "selection": "multiple",
        "status": "ready",
        "syncChildrenRemote": [Function],
        "totalCount": 0,
      }
    }
    mode="date"
    multiple={false}
    name="date"
    noValidate={false}
    placeholder={
      Array [
        "Start Date",
        "End Date",
      ]
    }
    readOnly={false}
    showHelp="newLine"
    suffixCls="calendar-picker"
    trigger={
      Array [
        "focus",
        "click",
      ]
    }
    triggerHiddenDelay={50}
    triggerShowDelay={150}
    trim="both"
  >
    <Tooltip
      arrowPointAtCenter={false}
      autoAdjustOverflow={true}
      defaultHidden={true}
      mouseEnterDelay={100}
      mouseLeaveDelay={100}
      placement="bottomLeft"
      suffixCls="tooltip"
      theme="light"
      title={null}
      transitionName="zoom-big-fast"
      trigger={
        Array [
          "hover",
        ]
      }
    >
      <Trigger
        action={
          Array [
            "hover",
          ]
        }
        arrowPointAtCenter={false}
        autoAdjustOverflow={true}
        blurDelay={0}
        builtinPlacements={
          Object {
            "bottom": Object {
              "offset": Array [
                0,
                10,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tc",
                "bc",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "bottomLeft": Object {
              "offset": Array [
                0,
                10,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tl",
                "bl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "bottomRight": Object {
              "offset": Array [
                0,
                10,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tr",
                "br",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "left": Object {
              "offset": Array [
                -10,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "cr",
                "cl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "leftBottom": Object {
              "offset": Array [
                -10,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "br",
                "bl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "leftTop": Object {
              "offset": Array [
                -10,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tr",
                "tl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "right": Object {
              "offset": Array [
                10,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "cl",
                "cr",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "rightBottom": Object {
              "offset": Array [
                10,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "bl",
                "br",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "rightTop": Object {
              "offset": Array [
                10,
                0,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tl",
                "tr",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "top": Object {
              "offset": Array [
                0,
                -10,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "bc",
                "tc",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "topLeft": Object {
              "offset": Array [
                0,
                -10,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "bl",
                "tl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
            "topRight": Object {
              "offset": Array [
                0,
                -10,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "br",
                "tr",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            },
          }
        }
        defaultPopupHidden={true}
        focusDelay={150}
        mouseEnterDelay={100}
        mouseLeaveDelay={100}
        popupContent={null}
        popupPlacement="bottomLeft"
        prefixCls="c7n-pro-tooltip"
        suffixCls="tooltip"
        theme="light"
        title={null}
        transitionName="zoom-big-fast"
      >
        <TriggerChild
          key=".$.$element"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <span
            className="c7n-pro-calendar-picker-wrapper c7n-pro-calendar-picker-required c7n-pro-calendar-picker-suffix-button c7n-pro-calendar-picker-range"
            key=".$element"
            onBlur={[Function]}
            onClick={[Function]}
            onContextMenu={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <label
              onMouseDown={[Function]}
            >
              <Trigger
                action={
                  Array [
                    "focus",
                    "click",
                  ]
                }
                blurDelay={50}
                builtinPlacements={
                  Object {
                    "bottomLeft": Object {
                      "offset": Array [
                        0,
                        4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tl",
                        "bl",
                      ],
                    },
                    "bottomRight": Object {
                      "offset": Array [
                        0,
                        4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tr",
                        "br",
                      ],
                    },
                    "topLeft": Object {
                      "offset": Array [
                        0,
                        -4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "bl",
                        "tl",
                      ],
                    },
                    "topRight": Object {
                      "offset": Array [
                        0,
                        -4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "br",
                        "tr",
                      ],
                    },
                  }
                }
                defaultPopupHidden={true}
                focusDelay={150}
                getPopupStyleFromAlign={[Function]}
                getRootDomNode={[Function]}
                mouseEnterDelay={150}
                mouseLeaveDelay={50}
                onPopupAnimateAppear={[Function]}
                onPopupAnimateEnd={[Function]}
                onPopupHiddenChange={[Function]}
                popupContent={
                  <DaysView
                    date={"2020-01-21T16:00:00.000Z"}
                    isValidDate={[Function]}
                    mode="date"
                    onSelect={[Function]}
                    onSelectedDateChange={[Function]}
                    onViewModeChange={[Function]}
                    suffixCls="calendar"
                  />
                }
                popupHidden={true}
                popupPlacement="bottomLeft"
                prefixCls="c7n-pro-calendar-picker"
                transitionName="slide-up"
              >
                <TriggerChild
                  key=".$text"
                  onBlur={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                >
                  <span
                    className="c7n-pro-calendar-picker"
                    key="text"
                    onBlur={[Function]}
                    onClick={[Function]}
                    onContextMenu={[Function]}
                    onFocus={[Function]}
                    onMouseDown={[Function]}
                    onMouseEnter={[Function]}
                    onMouseLeave={[Function]}
                  >
                    <span
                      className="c7n-pro-calendar-picker-range-text"
                      key="text"
                    >
                      <input
                        className="c7n-pro-calendar-picker-range-start"
                        onChange={[Function]}
                        onMouseDown={[Function]}
                        placeholder="Start Date"
                        readOnly={true}
                        tabIndex={-1}
                        value="1984-11-22"
                      />
                      <span
                        className="c7n-pro-calendar-picker-range-split"
                      >
                        ~
                      </span>
                      <input
                        className="c7n-pro-calendar-picker-range-end"
                        onChange={[Function]}
                        onMouseDown={[Function]}
                        placeholder="End Date"
                        readOnly={true}
                        tabIndex={-1}
                        value="2020-02-09"
                      />
                      <input
                        autoComplete="off"
                        className="c7n-pro-calendar-picker-range-input"
                        key="text"
                        multiple={false}
                        name="date"
                        noValidate={false}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        onKeyDown={[Function]}
                        placeholder=""
                        readOnly={false}
                        style={
                          Object {
                            "left": 0,
                          }
                        }
                        type="text"
                        value=""
                      />
                    </span>
                  </span>
                </TriggerChild>
                <Popup
                  align={
                    Object {
                      "offset": Array [
                        0,
                        4,
                      ],
                      "overflow": Object {
                        "adjustX": 1,
                        "adjustY": 1,
                      },
                      "points": Array [
                        "tl",
                        "bl",
                      ],
                    }
                  }
                  className="c7n-pro-calendar-picker-popup"
                  getClassNameFromAlign={[Function]}
                  getRootDomNode={[Function]}
                  getStyleFromAlign={[Function]}
                  hidden={true}
                  key="popup"
                  onAnimateAppear={[Function]}
                  onAnimateEnd={[Function]}
                  onMouseDown={[Function]}
                  suffixCls="popup"
                  transitionName="slide-up"
                />
              </Trigger>
              <div
                className="c7n-pro-calendar-picker-inner-button c7n-pro-calendar-picker-clear-button"
                key="inner-button"
              >
                <Icon
                  onClick={[Function]}
                  type="close"
                >
                  <i
                    className="icon icon-close"
                    onClick={[Function]}
                  />
                </Icon>
              </div>
              <div
                className="c7n-pro-calendar-picker-suffix"
                onMouseDown={[Function]}
              >
                <Icon
                  className="c7n-pro-calendar-picker-trigger"
                  type="date_range"
                >
                  <i
                    className="icon icon-date_range c7n-pro-calendar-picker-trigger"
                  />
                </Icon>
              </div>
            </label>
          </span>
        </TriggerChild>
        <Popup
          align={
            Object {
              "offset": Array [
                0,
                10,
              ],
              "overflow": Object {
                "adjustX": 1,
                "adjustY": 1,
              },
              "points": Array [
                "tl",
                "bl",
              ],
              "targetOffset": Array [
                0,
                0,
              ],
            }
          }
          className="c7n-pro-tooltip-popup"
          getClassNameFromAlign={[Function]}
          getRootDomNode={[Function]}
          hidden={true}
          key="popup"
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          suffixCls="popup"
          transitionName="zoom-big-fast"
        />
      </Trigger>
    </Tooltip>
  </DatePicker>
</DatePickerTest>
`;

exports[`date-picker-pro renders datePicker correctly 1`] = `
<DatePicker
  autoComplete="off"
  clearButton={true}
  mode="date"
  multiple={false}
  noValidate={false}
  readOnly={false}
  showHelp="newLine"
  suffixCls="calendar-picker"
  trigger={
    Array [
      "focus",
      "click",
    ]
  }
  triggerHiddenDelay={50}
  triggerShowDelay={150}
  trim="both"
>
  <Tooltip
    arrowPointAtCenter={false}
    autoAdjustOverflow={true}
    defaultHidden={true}
    mouseEnterDelay={100}
    mouseLeaveDelay={100}
    placement="bottomLeft"
    suffixCls="tooltip"
    theme="light"
    title={null}
    transitionName="zoom-big-fast"
    trigger={
      Array [
        "hover",
      ]
    }
  >
    <Trigger
      action={
        Array [
          "hover",
        ]
      }
      arrowPointAtCenter={false}
      autoAdjustOverflow={true}
      blurDelay={0}
      builtinPlacements={
        Object {
          "bottom": Object {
            "offset": Array [
              0,
              10,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tc",
              "bc",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "bottomLeft": Object {
            "offset": Array [
              0,
              10,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tl",
              "bl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "bottomRight": Object {
            "offset": Array [
              0,
              10,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tr",
              "br",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "left": Object {
            "offset": Array [
              -10,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "cr",
              "cl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "leftBottom": Object {
            "offset": Array [
              -10,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "br",
              "bl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "leftTop": Object {
            "offset": Array [
              -10,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tr",
              "tl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "right": Object {
            "offset": Array [
              10,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "cl",
              "cr",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "rightBottom": Object {
            "offset": Array [
              10,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "bl",
              "br",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "rightTop": Object {
            "offset": Array [
              10,
              0,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tl",
              "tr",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "top": Object {
            "offset": Array [
              0,
              -10,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "bc",
              "tc",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "topLeft": Object {
            "offset": Array [
              0,
              -10,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "bl",
              "tl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
          "topRight": Object {
            "offset": Array [
              0,
              -10,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "br",
              "tr",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          },
        }
      }
      defaultPopupHidden={true}
      focusDelay={150}
      mouseEnterDelay={100}
      mouseLeaveDelay={100}
      popupContent={null}
      popupPlacement="bottomLeft"
      prefixCls="c7n-pro-tooltip"
      suffixCls="tooltip"
      theme="light"
      title={null}
      transitionName="zoom-big-fast"
    >
      <TriggerChild
        key=".$.$element"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
      >
        <span
          className="c7n-pro-calendar-picker-wrapper c7n-pro-calendar-picker-empty c7n-pro-calendar-picker-suffix-button"
          key=".$element"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
        >
          <label
            onMouseDown={[Function]}
          >
            <Trigger
              action={
                Array [
                  "focus",
                  "click",
                ]
              }
              blurDelay={50}
              builtinPlacements={
                Object {
                  "bottomLeft": Object {
                    "offset": Array [
                      0,
                      4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "tl",
                      "bl",
                    ],
                  },
                  "bottomRight": Object {
                    "offset": Array [
                      0,
                      4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "tr",
                      "br",
                    ],
                  },
                  "topLeft": Object {
                    "offset": Array [
                      0,
                      -4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "bl",
                      "tl",
                    ],
                  },
                  "topRight": Object {
                    "offset": Array [
                      0,
                      -4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "br",
                      "tr",
                    ],
                  },
                }
              }
              defaultPopupHidden={true}
              focusDelay={150}
              getPopupStyleFromAlign={[Function]}
              getRootDomNode={[Function]}
              mouseEnterDelay={150}
              mouseLeaveDelay={50}
              onPopupAnimateAppear={[Function]}
              onPopupAnimateEnd={[Function]}
              onPopupHiddenChange={[Function]}
              popupContent={
                <DaysView
                  date={"2020-01-21T16:00:00.000Z"}
                  isValidDate={[Function]}
                  mode="date"
                  onSelect={[Function]}
                  onSelectedDateChange={[Function]}
                  onViewModeChange={[Function]}
                  suffixCls="calendar"
                />
              }
              popupHidden={true}
              popupPlacement="bottomLeft"
              prefixCls="c7n-pro-calendar-picker"
              transitionName="slide-up"
            >
              <TriggerChild
                key=".$text"
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
              >
                <input
                  autoComplete="off"
                  className="c7n-pro-calendar-picker"
                  key="text"
                  multiple={false}
                  noValidate={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onContextMenu={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  onMouseDown={[Function]}
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                  readOnly={false}
                  type="text"
                  value=""
                />
              </TriggerChild>
              <Popup
                align={
                  Object {
                    "offset": Array [
                      0,
                      4,
                    ],
                    "overflow": Object {
                      "adjustX": 1,
                      "adjustY": 1,
                    },
                    "points": Array [
                      "tl",
                      "bl",
                    ],
                  }
                }
                className="c7n-pro-calendar-picker-popup"
                getClassNameFromAlign={[Function]}
                getRootDomNode={[Function]}
                getStyleFromAlign={[Function]}
                hidden={true}
                key="popup"
                onAnimateAppear={[Function]}
                onAnimateEnd={[Function]}
                onMouseDown={[Function]}
                suffixCls="popup"
                transitionName="slide-up"
              />
            </Trigger>
            <div
              className="c7n-pro-calendar-picker-inner-button c7n-pro-calendar-picker-clear-button"
              key="inner-button"
            >
              <Icon
                onClick={[Function]}
                type="close"
              >
                <i
                  className="icon icon-close"
                  onClick={[Function]}
                />
              </Icon>
            </div>
            <div
              className="c7n-pro-calendar-picker-suffix"
              onMouseDown={[Function]}
            >
              <Icon
                className="c7n-pro-calendar-picker-trigger"
                type="date_range"
              >
                <i
                  className="icon icon-date_range c7n-pro-calendar-picker-trigger"
                />
              </Icon>
            </div>
          </label>
        </span>
      </TriggerChild>
      <Popup
        align={
          Object {
            "offset": Array [
              0,
              10,
            ],
            "overflow": Object {
              "adjustX": 1,
              "adjustY": 1,
            },
            "points": Array [
              "tl",
              "bl",
            ],
            "targetOffset": Array [
              0,
              0,
            ],
          }
        }
        className="c7n-pro-tooltip-popup"
        getClassNameFromAlign={[Function]}
        getRootDomNode={[Function]}
        hidden={true}
        key="popup"
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        suffixCls="popup"
        transitionName="zoom-big-fast"
      />
    </Trigger>
  </Tooltip>
</DatePicker>
`;
